crawler/jobs/cherrypicker_test_data.py

```diff
@@ -0,0 +1,186 @@
+    plate will contain an exact number of positive results between 0 and 96 as
+    specified. Up to 200 plates can be generated at a time.
+
+    Arguments:
+        run_id: str - The ID of the run.  If this is not found in Mongo an
+            exception will be thrown.
+
+    Returns:
+        Metadata about the plates generated, as:
+        [ [ "barcode1", "description1" ], [ "barcode2", "description2" ] ]
+    """
+    logger.info("Begin generating data.")
+
+    if config is None:
+        config, _ = get_config()
+
+    with create_mongo_client(config) as mongo_client:
+        mongo_db = get_mongo_db(config, mongo_client)
+        collection = get_mongo_collection(mongo_db, COLLECTION_CHERRYPICK_TEST_DATA)
+
+        return process_run(config, collection, run_id)
+
+
+def process_run(config: Config, collection: Collection, run_id: str) -> List[List[str]]:
+    dt = datetime.utcnow()
+    run_doc = get_run_doc(collection, run_id)
+
+    if run_doc[FIELD_STATUS] != FIELD_STATUS_PENDING:
+        raise TestDataError(f"{TEST_DATA_ERROR_WRONG_STATE} '{FIELD_STATUS_PENDING}'")
+
+    try:
+        plate_specs, num_plates = extract_plate_specs(run_doc.get(FIELD_PLATE_SPECS))
+        add_to_dart = parse_bool_field(run_doc.get(FIELD_ADD_TO_DART), False)
+
+        update_status(collection, run_id, FIELD_STATUS_STARTED)
+        barcodes = create_barcodes(config, num_plates)
+
+        update_status(collection, run_id, FIELD_STATUS_PREPARING_DATA)
+        prepare_data(plate_specs, dt, barcodes, config.DIR_DOWNLOADED_DATA)
+
+        update_status(collection, run_id, FIELD_STATUS_CRAWLING_DATA)
+        run_crawler(sftp=False, keep_files=False, add_to_dart=add_to_dart, centre_prefix=TEST_DATA_CENTRE_PREFIX)
+
+        barcode_meta = create_barcode_meta(plate_specs, barcodes)
+        update_run(
+            collection,
+            run_id,
+            {
+                FIELD_STATUS: FIELD_STATUS_COMPLETED,
+                FIELD_BARCODES: json.dumps(barcode_meta),
+            },
+        )
+
+        return barcode_meta
+    except Exception as e:
+        update_run(
+            collection,
+            run_id,
+            {
+                FIELD_STATUS: FIELD_STATUS_FAILED,
+                FIELD_FAILURE_REASON: str(e),
+            },
+        )
+        raise
+
+
+def get_run_doc(collection, run_id):
+    logger.info(f"Getting Mongo document for ID: {run_id}")
+
+    run_doc = collection.find_one(ObjectId(run_id))
+    if run_doc is None:
+        raise TestDataError(f"{TEST_DATA_ERROR_NO_RUN_FOR_ID} '{run_id}'")
+    logger.debug(f"Found run: {run_doc}")
+
+    return run_doc
+
+
+def extract_plate_specs(plate_specs_string):
+    try:
+        plate_specs: List[List[int]] = json.loads(plate_specs_string)
+    except (TypeError, json.JSONDecodeError):
+        raise TestDataError(TEST_DATA_ERROR_INVALID_PLATE_SPECS)
+
+    num_plates = reduce(lambda a, b: a + b[0], plate_specs, 0)
+    if num_plates < 1 or num_plates > 200:
```

Comment:
The higher limit should be extracted to config instead to prevent having a code change to increase?